test: Box component snapshot replacements

dbchristopher opened the pull request:
### Requirements

Please check the following items are addressed in your pull request (or are not applicable)

- [ ] Includes test coverage for all changes
- [ ] Documentation updated
- [ ] i18n impacts
- [ ] a11y impacts

lukelooker commented on packages/components/src/Layout/Box/Box.test.tsx:
```diff
@@ -36,42 +36,6 @@ import {
-    assertSnapshot(<Box mt={['large', null, 'medium']}>🥑</Box>)
-  })
-
-  test('Box with SizeNone is valid', () => {
-    assertSnapshot(<Box mt="none">🥑</Box>)
-  })
-
-  test('Box supports background-color', () => {
-    assertSnapshot(<Box bg="key" />)
-  })
```
Might be good to use RTL to test that these are assigned as-expected.

dbchristopher replied:
Oh I just replaced these with storyshots, with screenshots taken at different breakpoints. That's probably sufficient to confirm that key color and responsive values are assigned as expected, right?

lukelooker replied:
My only concern is verifying that Box is getting fully "exercised" but I guess `Box` really is completely logic-less so there's nothing to test beyond "it doesn't crash". 

Ignore me :)

lukelooker commented on packages/components/src/Layout/Box/Box.test.tsx:
```diff
@@ -80,37 +44,6 @@ describe('Box', () => {
-  })
-
-  describe('borders', () => {
-    test('Box supports borders', () => {
-      assertSnapshot(<Box border="1px solid black" />)
-    })
-
-    test('supports borderRadius', () => {
-      assertSnapshot(<Box borderRadius="medium" />)
-    })
-
-    test('supports borderColor', () => {
-      assertSnapshot(<Box borderColor="ui2" />)
-    })
```
Since these are doing derived-theme behaviors I think simple RTL tests would be helpful here.

dbchristopher replied:
Same here -- I moved the border settings to storyshots. In general, anything style related we'll want to do in storyshots, right?

<img width="615" alt="Screen Shot 2021-02-11 at 9 28 49 AM" src="https://user-images.githubusercontent.com/238293/107674204-93072c80-6c4b-11eb-97b7-914d8c716469.png">

lukelooker commented on packages/components/src/Layout/Box/Box.test.tsx:
```diff
@@ -80,37 +44,6 @@ describe('Box', () => {
     assertSnapshot(<Box bottom="1rem" />)
```
I think this test could be replaced with RTL as well - poking at the `hasStyledRule`

dbchristopher replied:
I'm going to add more another storyshot to confirm absolute positioning

lukelooker replied:
👍

lukelooker commented on packages/components/src/Layout/Box/Box.test.tsx:
```diff
@@ -80,37 +44,6 @@ describe('Box', () => {
-
-  describe('fonts', () => {
-    test('font helpers', () => {
-      assertSnapshot(<Box fontFamily="brand" />)
-      assertSnapshot(<Box fontSize="small" />)
-      assertSnapshot(<Box fontWeight="bold" />)
-      assertSnapshot(<Box letterSpacing="10" />)
-      assertSnapshot(<Box lineHeight="medium" />)
-      assertSnapshot(<Box textAlign="right" />)
-      assertSnapshot(<Box color="ui4" />)
```
I think there's probably value in keeping the `fontFamily`, `fontSize`, `lineHeight` and `color` as RTL tests (although they could totally be done as one big one) to prove that theme values are pushing down as expected.

lukelooker replied:
I think you're right to dump these - really adding tests here would just be verifying that Styled System is working properly - maybe not a bad idea but not really `Box`'s responsibility

lukelooker reviewed (commented):
Happened across this PR - realized after digging in that perhaps it's not complete so feel free to ignore my comments as you see appropriate :)

dbchristopher commented:
ok @lukelooker this should be ready for final review! I moved all the style tests into storybook, and updated the UX tests to use RTL. 

As an aside, I have no idea why it triggered these <1 — 10 byte changes in some of the field components. Weird 🤷 

<img width="692" alt="Screen Shot 2021-02-11 at 3 50 37 PM" src="https://user-images.githubusercontent.com/238293/107713913-1b53f480-6c81-11eb-93b8-fec66f2c70c6.png">

lukelooker commented on packages/components/src/Layout/Box/Box.test.tsx:
```diff
@@ -25,146 +25,57 @@
-      const boxAsButton = mountWithTheme(<Box as="button" />)
-      expect(boxAsButton.find('button').exists()).toBeTruthy()
+      renderWithTheme(<Box as="button">Press Me</Box>)
+      expect(screen.getByText('Press Me').tagName).toEqual('BUTTON')
```
Could you use `getByRole` here?

```suggestion
      expect(screen.getByRole('button')).toBeInTheDocument()
```

dbchristopher replied:
oh yep, that's better! good call

lukelooker commented on packages/components/src/Layout/Box/Box.test.tsx:
```diff
@@ -25,146 +25,57 @@
+      renderWithTheme(
+        <Box as="input" type="text" defaultValue="Autofocus" autoFocus />
+      )
+      expect(screen.getByDisplayValue('Autofocus')).toHaveFocus()
```
Neat :)

lukelooker reviewed (approved):
LGTM. 

Made one minor pedantic suggestion for a jest-helper (feel free to ignore)

The snapshots with the 0 - 5 bytes changes are caused by some kind of "jitter" that happens when taking the snapshots in Github actions. I have a task to explore ignore those in the future because they are, admittedly annoying.

lukelooker reviewed (approved):
✅
